src/app/compiler/compiler-input.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+/*
+  opts:
+   - optimize
+   - { file_name: { library_name: address } }
+*/
+module.exports = (sources, opts) => {
+  return JSON.stringify({
+    language: 'Solidity',
+    sources,
+    settings: {
+      optimizer: {
+        enabled: opts.optimize ? opts.optimize : false,
```

Comment:
This seems like a not very useful construct, if you don't trust `optimize` to be a boolean you could do `enabled: opts.optimize === true`.